src/main/java/org/springframework/data/neo4j/repository/query/VectorSearchFragment.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2011-2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.data.neo4j.repository.query;
+
+import org.jspecify.annotations.Nullable;
+
+/**
+ * Collected params for vector search.
+ *
+ * @author Gerrit Meier
+ * @param indexName name of the index to use for vector search
+ * @param numberOfNodes number of nodes to fetch from the index search
+ * @param score score filter
+ */
+record VectorSearchFragment(String indexName, int numberOfNodes, @Nullable Double score) {
```

Comment:
Package-protected records FTW. We use these in a similar manner.

Nerd sniping: `OptionalDouble` 😵‍💫